### ISSUE TITLE
Enabler: Build Drupal Block for NCIDS Breadcrumb 

### ISSUE THREAD

**Issue by bryanpizzillo** [NCIOCPL/cgov-digital-platform#3242]
# Build Drupal Block for NCIDS Breadcrumb
> **ESTIMATE** 3

>_**NOTE:**_ We are able to use the existing Drupal back-end block for the breadcrumbs (see below). The only things that are outstanding on this ticket are:
>  * implementing the analytics
>  * 2 bugs where the page title portion of the breadcrumbs is breaking
>     * #3419
>       * #3366
>  * adding some way to support the current breadcrumbs on current templates between deployment of code and update of the web site to use the new theme. (This may only be an ODE issue right now)
> * The "Show" or "Hide" breadcrumb option needs to stay the same. We will update all of the microsites to have "Show" when launched.

The NCIDS breadcrumb markup differs from the current [CGDP Breadcrumbs](https://github.com/NCIOCPL/cgov-digital-platform/blob/master/docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/Plugin/Block/Breadcrumb.php).  We will be reusing the existing Breadcrumb block, but will update it to display the ?? title after the parent sections. This will modify the DCEG Connect site.

This should:
1. Use the wrapping variant
2. For the parent navigation items in the breadcrumbs, display the Navigation Label if it exists, otherwise display the Site Section Name.
3. For the current page you are on, display the page title.

### Acceptance Criteria
```gherkin
Scenario: User is on a c.gov site on desktop, viewing a L1 home and landing page
  Given they are on a L1 that uses the home and landing page content type (e.g. About Cancer)
  Then they will not see the breadcrumbs
```

```gherkin
Scenario: User is on a c.gov site on desktop, viewing a non-home-and-landing-page L1
  Given they are on an L1 that is any content type other than home and landing (e.g. Publications)
  Then they will see the breadcrumbs
    And the breadcrumbs will display {Root Site Section} > {Current Page Title}
    And {Root Site Section} will be a link to the landing page of the root site section
```
*Note*: The root site section name is for the language that you are viewing.

```gherkin
Scenario: User is on cancer.gov on desktop, viewing a L2
  Given they are on a L2 or lower (e.g. Adjusting to Cancer)
  Then they will see all breadcrumbs
    And the breadcrumbs will display starting with {Root Site Section}
    And {Root Site Section} will be a link to the landing page of the root site section
    And the breadcrumbs will display ending with the page they are on 
    And each level will be separated with a caret
    And each level will be a link to the landing page of the section
    But the page they are on will not be hyperlinked
```

```gherkin
Scenario: User is on a c.gov site on a mobile device
  Given they are on a c.gov site on a mobile device >480
  Then they will only see the parent of the page they are on
  And it will be hyperlinked
  And there will be an arrow to the left of it
```

### Acceptance Criteria for Analytics
``` gherkin
Given a user is on any page of cancer.gov other than home and landing pages
  And the user clicks on a link within the breadcrumbs section of the page
When the user clicks on a link in the mega menu
Then there should be an NCIDataLayer Other event 
  And the event has the name "Breadcrumbs:LinkClick"
  And the event has the the linkname "Breadcrumbs:LinkClick"
  And this event should have the following data:
      | key      | value                                                                                  |
      | linkText | Text of the link clicked                                                               |
      | location | Breadcrumbs                                                                            |
      | section  | The value of the hierarchy of the link clicked. ie:                                    |
      |          | If the top-level link is clicked, return L1. If the second link is clicked, return L2. |
```

--------

### Prerequisites
- [ ] https://github.com/NCIOCPL/ncids/issues/230
- [ ] #3241

**Comment by VictoriaSunNIH** [comment]
Add requirements for breadcrumbs related to site nav - business rules for when breadcrumbs show/don't show (microsites)

**Comment by VictoriaSunNIH** [comment]
@bryanpizzillo - I think this is a conversation to be had with Lynn.  Will add it to our check-in with her next Tuesday morning!

**Comment by bryanpizzillo** [comment]
The conversation was had. We will update DCEG Connect to show the current page.

**Comment by bryanpizzillo** [comment]
> The conversation was had. We will update DCEG Connect to show the current page.
Actually, we do not have to right now given the proposed implementation.

**Comment by bennettcc** [comment]
Per standup discussion on 7/22 - moving this to on hold to focus on header bug https://github.com/NCIOCPL/ncids/issues/512

**Comment by bennettcc** [comment]
2 SPs completed during S36, 1 remaining for S37

**Comment by sarinapadilla** [comment]
Closing this ticket as it was fixed by PR #3445!
